### DISCUSSION TITLE
how to get the meta info in middleware

### DISCUSSION THREAD

**Discussion by lamhungypl** [reduxjs/redux-toolkit#4301] [Q&A]
Hello folks, I'm using RtkQuery with endpoints.queryFn declarations and building a middleware to catch queryFn's `meta` [data](https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#adding-meta-information-to-queries). 
```typescript
  export const injectedApi = api.endpoints: (build) => ({
      Product: build.query<ANY, ANY>({
        queryFn: async () => {
          try {
          } catch (e){
            return { error: e, meta : { message: 'Error' }
          }
        },
      }),
    }),
```
Inspire from https://github.com/reduxjs/redux-toolkit/discussions/3432#discussioncomment-5859866 , I achieve the following result:

```typescript
type QueryMeta = {
  message?: boolean;
};
export const errorMetaDataMiddleware: Middleware =
  (api: MiddlewareAPI) => (next) => (action) => {
    if (isRejectedWithValue(action)) {
      if (isAnyOf(action.matchRejected)) {
        if (!action.meta?.baseQueryMeta?.message) {
          toast.error(message);
        }
      }
    }

    return next(action);
  };

```
This works, but the `action.meta?.baseQueryMeta` is not documented anywhere and is not type-safe.
My question is: are there any other utilities that help extract the `meta : { message: 'Error' }` in a type-safe way?

**Comment by markerikson** [comment]
No, you'll need to do some combination of existence checks and type guards.
